docs/source/development.rst

```diff
@@ -0,0 +1,104 @@
+
+At some point, coding examples in the master branch of the official repository may not work even with the latest release. In that case, users are recommended to either use the example code of the latest release or update the library code to the master branch.
+
+Branch strategy
+===============
+
+The official repository of Chainer Chemistry is https://github.com/pfnet-research/chainer-chemistry. 
+We use the *master* branch of the repository for development. Therefore, developer who makes PRs should send them to the master branch.
+
+During major version zero, we do not maintain any released versions.
+When a bug is found, changes for the bug should be merged to the next version (either minor or patch). If the bug is critical, we will release the next version as soon as possible.
+
+
+Coding guideline
+================
+
+We use upper camel case (e.g. ``FooBar``) for class names and snake case (e.g. ``foo_bar``) for function, method, variable and package names.
+
+
+Testing guideline
+=================
+
+Chainer Chemistry uses `pytest <https://docs.pytest.org/en/latest/index.html>`_  as a unit-test framework.
+All unit tests are located in ``tests/`` directory. We can run tests with normal usage of pytest.
+For example, the following command runs all unit tests::
+
+   $ pytest tests
+
+Some unit tests require GPUs, which are annotated with ``@pytest.mark.gpu``.
+Therefore, you can skip them with ``-m`` option::
+
+   $ pytest -m "not gpu" tests
+
+If a develop who write a unit test that uses GPUs, you must anotate it with ``@pytest.mark.gpu``.
+
+Similarly, some unit tests take long time to complete.
+We annotated them with ``@pytest.mark.slow`` and skip them with ``-m`` option::
+
+   $ pytest -m "not slow" tests
+
+If a develop who write a unit test that uses GPUs, you must anotate it with ``@pytest.mark.slow``.
+
+We can skip both GPU and slow tests with the following command::
+
+   $ pytest -m "not (gpu or slow)" tests
+
+
+Terminology
+===========
+
+In the context of machine learning, especially chemoinformatics, we use several terms such as feature, feature vectors, descriptor and so on
+to indicate representation of inputs. To avoid disambiguity and align naming convention within the library code, we use these terms in the following way:
+
+* *Feature* is a representation of a sample of interest (typically molecules in Chainer Chemistry).
+* *Label* is a target value of we want to predict.
+* *Input feature* is a representation of a sample from which we want to predict the target value.
+
+For example, consider a suepervised learning task whose dataset consisting of input-output pairs ``((x_1, y_1), ..., (x_N, y_N))``, where ``N`` is the number of samples.
+In Chainer Chemistry ``x_i` and ``y_i`` are called input feature and label, respectively and a pair of ``(x_i, y_i)`` is feature for each ``i``.
```

Comment:
and a pair of ``(x_i, y_i)`` is called feature for each ``i``.

Reply:
I intentionally omitted "call" to avoid repetition of call. I believed it is grammatically correct, but will ask native speakers which is more natural.